### PR TITLE
Add Saint Kitts and Nevis localizations for address entry

### LINKED ISSUE #38280: [Enhancement]: Add Saint Kitts and Nevis localizations for address entry
### Describe the solution you'd like

WC is currently missing the specifics for entering addresses in Saint Kitts and Nevis, which use named "Parishes".

Via support request (6272406-zd-woothemes), we received the following suggested addition:

 ```php
'KN' => array(
        'K' => __( 'Saint Kitts', 'woocommerce' ),
        'N' => __( 'Nevis', 'woocommerce' ),
        '01' => __( 'Christ Church Nichola Town', 'woocommerce' ),
        '02' => __( 'Saint Anne Sandy Point', 'woocommerce' ),
        '03' => __( 'Saint George Basseterre', 'woocommerce' ),
        '04' => __( 'Saint George Gingerland', 'woocommerce' ),
        '05' => __( 'Saint James Windward', 'woocommerce' ),
        '06' => __( 'Saint John Capisterre', 'woocommerce' ),
        '07' => __( 'Saint John Figtree', 'woocommerce' ),
        '08' => __( 'Saint Mary Cayon', 'woocommerce' ),
        '09' => __( 'Saint Paul Capisterre', 'woocommerce' ),
        '10' => __( 'Saint Paul Charlestown', 'woocommerce' ),
        '11' => __( 'Saint Peter Basseterre', 'woocommerce' ),
        '12' => __( 'Saint Thomas Lowland', 'woocommerce' ),
        '13' => __( 'Saint Thomas Middle Island', 'woocommerce' ),
        '15' => __( 'Trinity Palmetto Point', 'woocommerce' ),
    ),
```

We would also need to make a modification in https://github.com/woocommerce/woocommerce/blob/trunk/plugins/woocommerce/includes/class-wc-countries.php#L791 to support this properly.

### Describe alternatives you've considered

_No response_

### Additional context

_No response_

### PR DESCRIPTION
**Pull request by igorradovanov**
### Submission Review Guidelines:

- I have followed the [WooCommerce Contributing Guidelines](https://github.com/woocommerce/woocommerce/blob/trunk/.github/CONTRIBUTING.md) and the [WordPress Coding Standards](https://make.wordpress.org/core/handbook/best-practices/coding-standards/).
- I have checked to ensure there aren't other open [Pull Requests](https://github.com/woocommerce/woocommerce/pulls) for the same update/change.
- I have reviewed my code for [security best practices](https://developer.wordpress.org/apis/security/). 
- Following the above guidelines will result in quick merges and clear and detailed feedback when appropriate.

### Changes proposed in this Pull Request:

Closes #38280.

### How to test the changes in this Pull Request:

Using the [WooCommerce Testing Instructions Guide](https://github.com/woocommerce/woocommerce/wiki/Writing-high-quality-testing-instructions), include your detailed testing instructions:

1. Navigate WooCommerce -> Settings
2. Under Country/State you should see now Saint Kitts and Nevis parishes
3. Example: Search for _Christ Church Nichola Town_

### CODE REVIEW

File: plugins/woocommerce/includes/class-wc-countries.php [hunk 1]
```diff
@@ -1286,6 +1286,16 @@ public function get_country_locale() {
 							'priority' => 69,
 						),
 					),
+					'KN' => array(
+						'postcode' => array(
+							'required' => false,
+							'label' => __( 'Postal code', 'woocommerce' ),
```
Comment (rrennick) [inline plugins/woocommerce/includes/class-wc-countries.php]: The spacing on the `=>` needs to align with the `=>` on the previous line on both `label` properties added here

**Review by rrennick — CHANGES_REQUESTED** [review]
@igorradovanov Thanks for the help on this. I added the changelog entry for you. This should be ready to merge once the alignment is fixed on the array properties.

File: plugins/woocommerce/includes/class-wc-countries.php (hunk 1 above)
Comment (igorradovanov) [reply plugins/woocommerce/includes/class-wc-countries.php]: Thank you @rrennick, changes submitted

**Review by rrennick — APPROVED** [review]
@igorradovanov Thanks for the help.

**Comment by rrennick** [comment]
@igorradovanov Could you merge the latest `trunk` into this branch? We have some unrelated tests failing and merging is blocked.

**Comment by igorradovanov** [comment]
> @igorradovanov Could you merge the latest `trunk` into this branch? We have some unrelated tests failing and merging is blocked.

Completed, it should be fine by now

**Pull request merged**